src/corehost/cli/fxr_resolver.cpp

```diff
@@ -54,80 +53,90 @@ namespace
     }
 }
 
-bool resolve_fxr_path(const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
+bool fxr_resolver::try_get_path(const host_mode_t mode, const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
 {
     pal::string_t fxr_dir;
-#if FEATURE_APPHOST || FEATURE_LIBHOST
-    // If a hostfxr exists in app_root, then assume self-contained.
-    if (library_exists_in_dir(root_path, LIBFXR_NAME, out_fxr_path))
+    assert(mode != host_mode_t::invalid);
+    switch (mode)
     {
-        trace::info(_X("Resolved fxr [%s]..."), out_fxr_path->c_str());
-        out_dotnet_root->assign(root_path);
-        return true;
-    }
+        case host_mode_t::apphost:
+        case host_mode_t::libhost:
+        {
+            // If a hostfxr exists in app_root, then assume self-contained.
+            if (library_exists_in_dir(root_path, LIBFXR_NAME, out_fxr_path))
+            {
+                trace::info(_X("Resolved fxr [%s]..."), out_fxr_path->c_str());
+                out_dotnet_root->assign(root_path);
+                return true;
+            }
 
-    // For framework-dependent apps, use DOTNET_ROOT
+            // For framework-dependent apps, use DOTNET_ROOT
 
-    pal::string_t default_install_location;
-    pal::string_t dotnet_root_env_var_name = get_dotnet_root_env_var_name();
-    if (get_file_path_from_env(dotnet_root_env_var_name.c_str(), out_dotnet_root))
-    {
-        trace::info(_X("Using environment variable %s=[%s] as runtime location."), dotnet_root_env_var_name.c_str(), out_dotnet_root->c_str());
-    }
-    else
-    {
-        if (pal::get_dotnet_self_registered_dir(&default_install_location) || pal::get_default_installation_dir(&default_install_location))
-        {
-            trace::info(_X("Using global installation location [%s] as runtime location."), default_install_location.c_str());
-            out_dotnet_root->assign(default_install_location);
-        }
-        else
-        {
-            trace::error(_X("A fatal error occurred, the default install location cannot be obtained."));
-            return false;
-        }
-        trace::info(_X("Using default installation location [%s] as runtime location."), default_install_location.c_str());
-        out_dotnet_root->assign(default_install_location);
```

Comment:
Were these two lines intentionally rmoved/removed?

Reply:
Yes, they are (re)doing the same thing as in the if block above.